src/graphic/gl/utils.cc

```diff
@@ -266,7 +267,7 @@ void vertex_attrib_pointer(int vertex_index, int num_items, int stride, size_t o
 
 void swap_rows(const int width, const int height, const int pitch, const int bpp, uint8_t* pixels) {
 	uint8_t* begin_row = pixels;
-	uint8_t* end_row = pixels + pitch * (height - 1);
+	uint8_t* end_row = pixels + static_cast<ptrdiff_t>(pitch * (height - 1));
```

Comment:
i think casting the complete multiplication does not help as the potential error would happen before the cast